src/hevm/src/EVM/SMT.hs

```diff
@@ -35,16 +40,50 @@ import System.Process (createProcess, cleanupProcess, proc, ProcessHandle, std_i
 
 import EVM.Types
 import EVM.Expr hiding (copySlice, writeWord, op1, op2, op3, drop)
+import qualified Language.SMT2.Syntax as Language.SMT2.Parser
+import Language.SMT2.Syntax (SpecConstant(SCHexadecimal))
 
 
 -- ** Encoding ** ----------------------------------------------------------------------------------
+-- variable names in SMT that we want to get values for
+data CexVars = CexVars
+  { calldataV :: [Text]
+  , storageV :: Text
+  , blockContextV :: [Text]
+  , txContextV :: [Text]
+  }
+  deriving (Eq, Show)
+
+instance Semigroup CexVars where
+  (CexVars a b c d) <> (CexVars a2 b2 c2 d2) = CexVars (a <> a2) (b <> b2) (c <> c2) (d <> d2)
+
+instance Monoid CexVars where
+    mempty = CexVars
+      { calldataV = mempty
+      , storageV = mempty
+      , blockContextV = mempty
+      , txContextV = mempty
+      }
+
+data SMTCex = SMTCex
+  { calldata :: Map Text Language.SMT2.Parser.SpecConstant
+  , storage :: Text
```

Comment:
`storage`, `blockContext` and `txContext` are not currently used and I think it's fine to leave them as placeholders and add this functionality later.  I'm puzzled by their types though as their values is of `Text` type and not of `SpecConstant` type as calldata

Reply:
PR is here: https://github.com/ethereum/hevm/pull/62